includes/class-wc-payments.php

```diff
@@ -560,14 +560,24 @@ public static function get_plugin_headers() {
 	 */
 	public static function register_gateway( $gateways ) {
 		if ( WC_Payments_Features::is_upe_split_enabled() ) {
+
+			$payment_methods = self::$card_gateway->get_payment_method_ids_enabled_at_checkout();
+
+			if ( in_array( 'link', $payment_methods, true ) && self::$platform_checkout_button_handler->is_woopay_enabled() ) {
+				$key = array_search( 'link', $payment_methods, true );
```

Comment:
```suggestion
			$key = array_search( 'link', $payment_methods, true );

			if ( $key !== false && self::$platform_checkout_button_handler->is_woopay_enabled() ) {
```

As `array_search` returns false if a value doesn't exists in the array, wouldn't be better to use it to check if the key exists?